core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -341,12 +347,38 @@ public static String getJobWorkerMetricName(String name) {
    * @return the metric registry name
    */
   private static String getMetricNameWithUniqueId(InstanceType instance, String name) {
+    if (sSourceName == null) {
```

Comment:
Could you add the appropriate here to ensure `sSourceName` is only set once? Also do we assume each JVM will only have 1 instance type?

Reply:
Yeah, that's why assumed, each jvm will only have one instance type depends on the process type